src/Tribe/Tabbed_View.php

```diff
@@ -0,0 +1,366 @@
+		$slug = sanitize_title_with_dashes( $slug );
+
+		if ( ! empty( $this->items[ $slug ] ) ) {
+			return $this->items[ $slug ];
+		}
+
+		return null;
+	}
+
+	/**
+	 * Checks if a given Tab (slug) is active
+	 *
+	 * @param  string $slug The Slug of the Tab
+	 *
+	 * @return boolean       Is this tab active?
+	 */
+	public function is_active( $slug = null ) {
+		$slug = $this->get_requested_slug( $slug );
+		$tab  = $this->get_active();
+
+		return $slug === $tab->get_slug();
+	}
+
+	/**
+	 * Returns the slug of tab requested in the `_GET` array or the default one.
+	 *
+	 * @param string|null $slug
+	 * @param mixed       $default A default value to return if the tab was not requested.
+	 *
+	 * @return string|bool Either the slug of the requested tab or `false` if no slug was requested
+	 *                     and no default tab is set.
+	 */
+	protected function get_requested_slug( $slug = null, $default = null ) {
+		if ( is_null( $slug ) ) {
+			$default = null === $default ? $this->get_default_tab() : $default;
+			// Set the slug
+			$slug = ! empty( $_GET['tab'] ) && $this->exists( $_GET['tab'] ) ? $_GET['tab'] : $default;
+		}
+
+		return $slug;
+	}
+
+	/**
+	 * Fetches the current active tab instance.
+	 *
+	 * @return Tribe__Tabbed_View__Tab|bool The active tab, the default tab if no tab is active,
+	 *                                      `false` if no tabs are registered in the Tabbed View.
+	 */
+	public function get_active() {
+		if ( ! empty( $this->active ) && $this->exists( $this->active ) ) {
+			return $this->get( $this->active );
+		}
+
+		$tab = ! empty( $_GET['tab'] ) && $this->exists( $_GET['tab'] ) ? $_GET['tab'] : $this->get_default_tab();
+
+		// Return the active tab or the default one
+		return ! empty( $tab ) ? $this->get( $tab ) : false;
+	}
+
+	/**
+	 * Returns the slug of the default tab for this tabbed view.
+	 *
+	 * @return string The slug of the default tab, the slug of the first tab if
+	 *                a default tab is not set, `false` otherwise.
+	 */
+	public function get_default_tab() {
+		if ( ! empty( $this->default_tab ) && $this->exists( $this->default_tab ) ) {
+			return $this->default_tab;
+		}
+
+		$tabs = $this->get_tabs();
+
+		if ( empty( $tabs ) ) {
+			return false;
+		}
+
+		return reset( $tabs )->get_slug();
+	}
+
+	/**
+	 * @param Tribe__Tabbed_View__Tab|string $tab
+	 *
+	 * @return Tribe__Tabbed_View__Tab
+	 */
+	public function register( $tab ) {
+		$is_object = is_a( $tab, 'Tribe__Tabbed_View__Tab' );
+		if ( ! ( $is_object || ( is_string( $tab ) && class_exists( $tab ) ) ) ) {
+			return false;
+		}
+
+		if ( ! $is_object ) {
+			$tab = $this->get_new_tab_instance( $tab );
+		}
+
+		// Set the Tab Item on the array of Tabs
+		$tab_slug = $tab->get_slug();
+
+		if ( empty( $tab_slug ) ) {
+			return false;
+		}
+
+		$this->items[ $tab_slug ] = $tab;
+
+		// Return the tab
+		return $tab;
+	}
+
+	/**
+	 * Returns all the registered tabs.
+	 *
+	 * @return Tribe__Tabbed_View__Tab[]
```

Comment:
Wasn't actually aware that this was a valid way to express an array-of-a-certain-type in docblocks until now, nice!